ravendb/resource_ravendb_server.go

```diff
@@ -0,0 +1,385 @@
+package ravendb
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"net/http"
+	"sync"
+)
+
+const (
+	errorCreate = "error while creating information: %s"
+	errorRead   = "error getting raven configuration information: %s"
+	errorDelete = "error deleting RavenDbInstances: %s"
+)
+
+func resourceRavendbServer() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceServerCreate,
+		ReadContext:   resourceServerRead,
+		UpdateContext: resourceServerUpdate,
+		DeleteContext: resourceServerDelete,
+
+		Schema: map[string]*schema.Schema{
+			"hosts": {
+				Type:        schema.TypeList,
+				Required:    true,
+				Description: "The hostnames (or ip addresses) of the nodes that terraform will use to setup the RavenDB cluster.",
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validation.IsIPAddress,
+				},
+				MinItems: 1,
+			},
+			"healthcheck_database": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The database name to check whether he is alive or not.",
+			},
+			"certificate_base64": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "The cluster certificate file that is used by RavenDB for server side authentication.",
+			},
+			"license": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "The license that will be used for the setup of the RavenDB cluster.",
+				ValidateFunc: validation.StringIsBase64,
+			},
+			"version": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The RavenDB version to use for the cluster.",
+				ValidateFunc: func(val interface{}, key string) (warns []string, errs []error) {
+					v, ok := val.(string)
+					if !ok {
+						errs = append(errs, fmt.Errorf("expected type of %q to be string", v))
+						return warns, errs
+					}
+					link := "https://daily-builds.s3.us-east-1.amazonaws.com/ravendb_" + v + "-0_amd64.deb"
```

Comment:
We'll need to add `arch` (architecture) to determine which package to download too.